L1Trigger/L1CaloTrigger/plugins/Phase2L1CaloJetEmulator.cc

```diff
@@ -453,12 +655,271 @@ void Phase2L1CaloJetEmulator::produce(edm::Event& iEvent, const edm::EventSetup&
   iEvent.put(std::move(jetCands), "GCTJet");
 }
 
+// Apply calibrations to HCAL energy based on Jet Eta, Jet pT
+float Phase2L1CaloJetEmulator::get_jet_pt_calibration(float& jet_pt, float& jet_eta) const {
+  float abs_eta = std::abs(jet_eta);
+  float tmp_jet_pt = jet_pt;
+  if (tmp_jet_pt > 499)
+    tmp_jet_pt = 499;
+
+  // Different indices sizes in different calo regions.
+  // Barrel...
+  size_t eta_index = 0;
+  size_t pt_index = 0;
+  float calib = 1.0;
+  if (abs_eta <= 1.5) {
+    // Start loop checking 2nd value
+    for (unsigned int i = 1; i < absEtaBinsBarrel.size(); i++) {
+      if (abs_eta <= absEtaBinsBarrel.at(i))
+        break;
+      eta_index++;
+    }
+    // Start loop checking 2nd value
+    for (unsigned int i = 1; i < jetPtBins.size(); i++) {
+      if (tmp_jet_pt <= jetPtBins.at(i))
+        break;
+      pt_index++;
+    }
+    calib = calibrationsBarrel[eta_index][pt_index];
+  }                         // end Barrel
+  else if (abs_eta <= 3.0)  // HGCal
+  {
+    // Start loop checking 2nd value
+    for (unsigned int i = 1; i < absEtaBinsHGCal.size(); i++) {
+      if (abs_eta <= absEtaBinsHGCal.at(i))
+        break;
+      eta_index++;
+    }
+    // Start loop checking 2nd value
+    for (unsigned int i = 1; i < jetPtBins.size(); i++) {
+      if (tmp_jet_pt <= jetPtBins.at(i))
+        break;
+      pt_index++;
+    }
+    calib = calibrationsHGCal[eta_index][pt_index];
+  }     // end HGCal
+  else  // HF
+  {
+    // Start loop checking 2nd value
+    for (unsigned int i = 1; i < absEtaBinsHF.size(); i++) {
+      if (abs_eta <= absEtaBinsHF.at(i))
+        break;
+      eta_index++;
+    }
+    // Start loop checking 2nd value
+    for (unsigned int i = 1; i < jetPtBins.size(); i++) {
+      if (tmp_jet_pt <= jetPtBins.at(i))
+        break;
+      pt_index++;
+    }
+    calib = calibrationsHF[eta_index][pt_index];
+  }  // end HF
+
+  return jet_pt * calib;
+}
+
+// Apply calibrations to tau pT based on L1EG info, EM Fraction, Tau Eta, Tau pT
+float Phase2L1CaloJetEmulator::get_tau_pt_calibration(float& tau_pt, float& tau_eta) const {
```

Comment:
I think these can be constant references too

Reply:
Done